cmd/snap-confine/snap-confine.c

```diff
@@ -894,7 +898,49 @@ static void enter_non_classic_execution_environment(sc_invocation *inv, struct s
     // starting the snap application has already ensure that the process has
     // been put in a dedicated group.
     if (!sc_cgroup_is_v2()) {
```

Comment:
as discussed in a call, since this is a workaround for v1 only, we should have check like this early on, and when v2 is detected, SETUID/SETGID should be dropped from the permitted set

Reply:
Agreed. I'll post a follow-up.

Reply:
This is done